client/app/main/student/student.controller.js

```diff
@@ -2,9 +2,30 @@
 
 var app = angular.module('app');
 
-function StudentCtrl($scope, $state, $stateParams, Student, toastr) {
+function StudentCtrl($scope, $state, $stateParams, Student, 
+  AbsenceRecord, toastr) {
   $scope.student = Student.get({id: $stateParams.id});
 
+  $scope.calcPercentage = function() {
+    AbsenceRecord.current()
+      .$promise.then(function(records) {
+        var totPresent = [];
+        var totEnrolled = [];
+        var addToNext = function(prev, cur) {
+          return prev + cur;
+        };
+        records.forEach(function(record) {
+          totPresent.push(record.entries.present);
+          totEnrolled.push(record.entries.enrolled);
+        });
+        $scope.percentage = Math.floor(
+          (totPresent.reduce(addToNext) / totEnrolled.reduce(addToNext))
+          * 100);
+      });
+  };
+
+  $scope.calcPercentage();
+
```

Comment:
No. It probably needs to be added if we want data visualization for students (absences/tardies over  time). 

Using the AbsenceRecord.current(), while very fragile (if the student isn't in the current report it would break) does return the needed information. You would need to find the student from the list of all students.